### PR TITLE
Checkout: abstracting Brazil/Ebank form fields

### PR DESCRIPTION
**Pull request by ramonjd**
This PR extracts extra fields required for Ebanx payments in Brazil from the credit card fields component, in order that we may use them in other payment-related forms such as bank transfer/redirect.

Along with the fields, I'm also moving clientside field validation for checkout/credit card fields into `lib/checkout` so that we can extend the rule to include the paypal/redirect panels and so on.

There is no new functionality.

For a background see: `pxLjZ-4BN-p2`

## Screenshot

screen shot 2018-04-12 at 4 45 49 pm

## Testing
1. Set your currency to 'BRL'
2. Add something to your cart and head to checkout
3. At the payment step, select 'Brazil' as your country
4. Without entering any data, attempt to submit
5. Fill in credit card information (use a dummy card no#) and Ebanx fields (a valid tax code is `111.444.777-35`)

### Expectations
**At 3**: You should see the Ebanx extra fields
**At 4**: The fields should show validation messages 
**At 5**: Valid form data should permit payment attempt

### CODE REVIEW

**Comment by yoavf** [comment]
Works well, but I've noticed a small visual regression with the state/postal code fields.

**Before:**
screen shot 2018-04-12 at 11 12 35

**After:**
screen shot 2018-04-12 at 11 12 29

**Comment by ramonjd** [comment]
Thanks for catching that @yoavf 

I didn't pull the existing css over to the new component. Now looks like this:

_**Mobile**_

screen shot 2018-04-13 at 12 47 51 pm

_**Desktop**_

screen shot 2018-04-13 at 12 47 31 pm

**Comment by ramonjd** [comment]
Just realized after looking how to introduction validation to the redirect payment box that I need to abstract the [ebanx validation rules](https://github.com/Automattic/wp-calypso/blob/75452b5/client/lib/credit-card-details/validation.js
) as well. More to come on this PR... :)

File: client/lib/checkout/test/validation.js [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+/**
+ * External dependencies
+ */
+import moment from 'moment';
+
+/**
+ * Internal dependencies
+ */
+import { validatePaymentDetails, getCreditCardType } from '../validation';
+import { isEbanxEnabledForCountry, isValidCPF } from 'lib/checkout/ebanx';
```
[the remark this answers is not in the document]
Comment (ramonjd) [reply client/lib/checkout/test/validation.js]: Bit of a mouthful, but the least prolix I could come up with was: `isEbanxCreditCardProcessingEnabledForCountry`

**Comment by ramonjd** [comment]
> This works well, but I think we need to go a little bit deeper and cleanup some of the intermediate code, maybe in a way combine some of shouldRenderAdditionalEbanxFields, isEbanxEnabledForCountry and ebanxFieldRules. 

Ebanx is now peppered about the shop.

By 'combine' do you mean in the same module, or aggregate some of the logic into higher-order functions?

**Comment by ramonjd** [comment]
I've pulled together the common Ebanx methods, and genericized the `<EbanxPaymentFields />` in e83f842294bbdbeef04d05487ef89634b8e93e6e. 

Let me know if you see any further areas that could be cleaned up.

**Pull request merged**